File: exercises/beecrowd/1185/1185.ts
// https://judge.beecrowd.com/pt/problems/view/1185
import * as fs from "fs";

function main(): void {
  const input = fs.readFileSync("/dev/stdin", "utf8").trim().split("\n");
  const op = input[0];
  const numbers = input.slice(1).map(Number);

  let sum = 0;
  let count = 0;

  for (let i = 0; i < 12; i++) {
    for (let j = 0; j < 12; j++) {
      if (i + j < 11) {
        sum += numbers[i * 12 + j];
        count++;
      }
    }
  }

  if (op === "M") {
    sum /= count;
  }

  console.log(sum.toFixed(1));
}

main();

File: exercises/beecrowd/1185/1185.go
// https://judge.beecrowd.com/pt/problems/view/1185
package main

import "fmt"

func main() {
	var op string
	fmt.Scan(&op)

	matrix := make([][]float64, 12)
	for i := 0; i < 12; i++ {
		matrix[i] = make([]float64, 12)
		for j := 0; j < 12; j++ {
			fmt.Scan(&matrix[i][j])
		}
	}

	sum := 0.0
	count := 0

	for i := 0; i < 12; i++ {
		for j := 0; j < 12; j++ {
			if i+j < 11 {
				sum += matrix[i][j]
				count++
			}
		}
	}

	if op == "M" {
		sum /= float64(count)
	}

	fmt.Printf("%.1f\n", sum)
}
